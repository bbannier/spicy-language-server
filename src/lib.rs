#![feature(str_strip)]

#[macro_use]
extern crate pest;

pub mod lsp;
pub mod options;
pub mod spicy;
