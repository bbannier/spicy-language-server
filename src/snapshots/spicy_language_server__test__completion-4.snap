---
source: src/lib.rs
expression: "server.completion(CompletionParams\n{\n    text_document_position:\n    TextDocumentPositionParams::new(TextDocumentIdentifier::new(uri.clone()),\n    Position::new(1, 0),), work_done_progress_params:\n    WorkDoneProgressParams::default(), partial_result_params:\n    PartialResultParams::default(), context: None,\n}).await"
---
Ok(
    Some(
        Array(
            [
                CompletionItem {
                    label: "unit",
                    label_details: None,
                    kind: Some(
                        Snippet,
                    ),
                    detail: None,
                    documentation: None,
                    deprecated: None,
                    preselect: None,
                    sort_text: None,
                    filter_text: None,
                    insert_text: Some(
                        "type ${1:X} = unit {\n    ${2:a}: ${3:uint8};\n};\n",
                    ),
                    insert_text_format: Some(
                        Snippet,
                    ),
                    insert_text_mode: None,
                    text_edit: None,
                    additional_text_edits: None,
                    command: None,
                    commit_characters: None,
                    data: None,
                    tags: None,
                },
                CompletionItem {
                    label: "unit",
                    label_details: None,
                    kind: Some(
                        Keyword,
                    ),
                    detail: None,
                    documentation: None,
                    deprecated: None,
                    preselect: None,
                    sort_text: None,
                    filter_text: None,
                    insert_text: Some(
                        "unit",
                    ),
                    insert_text_format: None,
                    insert_text_mode: None,
                    text_edit: None,
                    additional_text_edits: None,
                    command: None,
                    commit_characters: None,
                    data: None,
                    tags: None,
                },
            ],
        ),
    ),
)
